src/mito2/src/worker.rs

```diff
@@ -629,6 +667,24 @@ impl<S: LogStore> RegionWorkerLoop<S> {
         }
     }
 
+    /// Handle periodical tasks such as region auto flush.
+    fn handle_periodical_tasks(&mut self) {
+        let interval = CHECK_REGION_INTERVAL.as_millis() as i64;
+        if self
+            .time_provider
+            .elapsed_since(self.last_periodical_check_millis)
+            < interval
+        {
+            return;
+        }
+
+        if let Err(e) = self.flush_periodically() {
+            error!(e; "Failed to flush regions periodically");
+        }
+
+        self.last_periodical_check_millis = self.time_provider.current_time_millis();
```

Comment:
How about move this line above the `flush_periodically` call? If that method runs too long (by accidently unawarely  adding more feature), the interval to run flush might be incorrect.

Reply:
fixed in https://github.com/GreptimeTeam/greptimedb/pull/3459/commits/a4145cafa56b2fff866ae19a43d681ffca0319f9